docs/RFCS/range_leases.md

```diff
@@ -47,28 +47,20 @@ its replicas holding range leases at once?
 
 We introduce a new node lease table at the beginning of the keyspace
 (not an actual SQL table; it will need to be accessed with lower-level
-APIs). This table is special in several respects: it is gossipped, and
+APIs). This table is special in several respects: it is gossiped, and
 leases within its keyspace (and all ranges that precede it, including
 meta1 and meta2) use the current, per-range lease mechanism to avoid
 circular dependencies. This table maps node IDs to an epoch counter
-and a lease expiration timestamp.
-
-The range lease is moved from a special raft command (which writes to
-a range-local, non-transactional range lease key) to a transactional
-range-local key (similar to the range descriptor). The range lease
-identifies the node that holds the lease and its epoch counter. It has
-a start timestamp but does not include an expiration time. The range
-lease record is always updated in a distributed transaction with the
-node lease record to ensure that the epoch counter is consistent and
-the start time is greater than the prior range lease holder’s node
-lease expiration (plus the maximum clock offset).
+and a node heartbeat timestamp.
 
 Each node periodically performs a conditional put to its node lease to
-increase the expiration timestamp and ensure that the epoch has not
+increase the heartbeat timestamp and ensure that the epoch has not
 changed. If the epoch does change, *all* of the range leases held by
 this node are revoked. A node *must not* propose any commands with a
-timestamp greater than the latest expiration timestamp it has written
-to the node lease table.
+timestamp greater than the latest heartbeat timestamp plus a liveness
+threshold duration. Note that the heartbeat is specified according to
```

Comment:
I think you are trying to express something like spanner's "disjointed invariant" here. It will be better to say that the lease expires after the heartbeat-timestamp + liveness-threshold-duration . Not clear what this liveness-threshold-duration is?